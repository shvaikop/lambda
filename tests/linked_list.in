true = \x.\y.x
false = \x.\y.y
pair = \x.\y.\f.f x y
first = \p.p $true
second = \p.p $false
nil = \x.$true
null = \p.p (\x.\y.$false)
zero = \f.\x.x
one = \f.\x.f x
two = \f.\x.f (f x)
three = \f.\x.f (f (f x))
a = $pair $three $nil
b = $pair $two $a
c = $pair $one $b
head = $pair $zero $c
$first $head
$first ($second $head)
$first (  $second ($second $head)   )
$first ($second ($second ($second $head)))
