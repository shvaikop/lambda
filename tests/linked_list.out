Up to date
Welcome to Lambda Calculus Interpreter
\f.\x.x
\f.\x.f x
\f.\x.(f) (f x)
\f.\x.(f) ((f) (f x))
